swift/ql/lib/codeql/swift/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -60,7 +61,25 @@ private module Cached {
   cached
   newtype TNode =
     TExprNode(ExprCfgNode e) or
-    TSsaDefinitionNode(Ssa::Definition def)
+    TSsaDefinitionNode(Ssa::Definition def) or
+    TInoutReturnNode(ParamDecl param, ControlFlowNode exit) {
```

Comment:
So if a function has `n` inout parameters and `m` return statements then we'll have `n * m` `TInoutReturnNode`s? That sounds a bit scary. Do we really need the `exit` column here? I would imagine we could just flow from the last ref of the `inout` parameter (using the `lastRef` maybe?) to a `InoutReturnNode` node that's indexed by the `param` column only.

Reply:
> Do we really need the exit column here?

I don't think so, but I believe we'd need extra CFG nodes to make that work.